applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp

```diff
@@ -416,6 +416,134 @@ bool CheckWakeCondition(const Element& rElement, const double& rTolerance, const
+
+    // maximum local squared mach number
+    const double sq_max_local_mach_number = rCurrentProcessInfo[MACH_SQUARED_LIMIT];
+
+    // read free stream values
+    const double heat_capacity_ratio = rCurrentProcessInfo[HEAT_CAPACITY_RATIO];
+    const double free_stream_mach = rCurrentProcessInfo[FREE_STREAM_MACH];
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+
+    // make squares of values
+    double sq_free_stream_mach = std::pow(free_stream_mach, 2);
+    double sq_free_stream_velocity = inner_prod(free_stream_velocity, free_stream_velocity);
+
+    return sq_max_local_mach_number * sq_free_stream_velocity * 
+            ((sq_free_stream_mach * heat_capacity_ratio - sq_free_stream_mach + 2) / 
+            (sq_free_stream_mach * heat_capacity_ratio * sq_max_local_mach_number 
+            - sq_max_local_mach_number * sq_free_stream_mach + 2 * sq_free_stream_mach));
+}
+
+template <int Dim, int NumNodes>
+double ComputeClampedVelocitySquared(
+    const array_1d<double, Dim>& rVelocity, 
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // check if velocity is over max allowed velocity
+    double sq_max_velocity = ComputeMaximumVelocitySquared<Dim, NumNodes>(rCurrentProcessInfo);
+
+    double sq_local_velocity = inner_prod(rVelocity, rVelocity);
+
+    if (sq_local_velocity > sq_max_velocity)
+    {
+        KRATOS_WARNING("Clamped local velocity") << 
+        "SQUARE OF LOCAL VELOCITY ABOVE ALLOWED SQUARE OF VELOCITY"
+        << " sq_local_velocity  = " << sq_max_velocity
+        << " sq_max_velocity  = " << sq_max_velocity << std::endl;
+
+        sq_local_velocity = sq_max_velocity;
+    }
+
+    return sq_local_velocity;
+}
+
+template <int Dim, int NumNodes>
+double ComputeLocalSpeedofSoundSquared(
+    const array_1d<double, Dim>& rVelocity,
+    const ProcessInfo& rCurrentProcessInfo)
+{
+    // Implemented according to Equation 8.7 of Drela, M. (2014) Flight Vehicle
+    //      Aerodynamics, The MIT Press, London
+
+    // read free stream values
+    const double heat_capacity_ratio = rCurrentProcessInfo[HEAT_CAPACITY_RATIO];
+    const double free_stream_mach = rCurrentProcessInfo[FREE_STREAM_MACH];
+    const array_1d<double, 3> free_stream_velocity = rCurrentProcessInfo[FREE_STREAM_VELOCITY];
+    const double free_stream_speed_sound = rCurrentProcessInfo[SOUND_VELOCITY];
+
+    // make squares of values
+    double sq_free_stream_mach = std::pow(free_stream_mach, 2);
+    double sq_free_stream_speed_sound = std::pow(free_stream_speed_sound,2);
+    double sq_free_stream_velocity = inner_prod(free_stream_velocity, free_stream_velocity);
+
+    // computes square of velocity including clamping according to MACH_SQUARED_LIMIT
+    double sq_local_velocity = ComputeClampedVelocitySquared<Dim, NumNodes>(rVelocity, rCurrentProcessInfo);
+
+    // square bracket term
+    double square_bracket_term = 1.0 + 0.5*(heat_capacity_ratio - 1.0)*
```

Comment:
Just a thought, I saw that you also compute the square brackets term in the ComputeDerivaticeMachSquaredWRTVelocitySquared. I will also pop up in the computation of the density. And actually it appears in many other places. Could you please include a separate function CompureSquareBracketsTerm passing the local velocity squared and the process info? I think it makes sense, doesn't it?

Reply:
@Inigo Yes, I think that makes sense as this term appears a lot and will appear also in upcoming functions. However, is there a more meaningful name for this term? It seems following Drela it originally comes from the enthalpy:
![Screen Shot 2020-04-23 at 3 20 47 PM](https://user-images.githubusercontent.com/60355972/80103923-0b461980-8576-11ea-8556-ac88a7997953.png)

Reply:
You are right, we should think of a more meaningful name. I have always referred to it as square brackets term cause Eller does so. However if you are looking for its physical meaning, from equation 8.7:

![a](https://user-images.githubusercontent.com/28628414/80114252-a7c2e880-8583-11ea-9ebd-ecaa67fa7f89.png)

you can see that the square brackets term (in this case round brackets term) is nothing else than the ratio between the local speed of sound and the free stream speed of sound (squared). You can also see that in your implementation:

https://github.com/KratosMultiphysics/Kratos/blob/6ed32c132aef9e7fa78e784cbc612f3503a53cb5/applications/CompressiblePotentialFlowApplication/custom_utilities/potential_flow_utilities.cpp#L526


Shall we call it speed_of_sound_ratio_squared?

Let us discuss it tomorrow, just leave it as you have it now (without making a new function) because it is already good and I feel like I just overthought about it yesterday.

Reply:
At the end of the day it is just a double and a line of code, it might be too much to make a function just for that

Reply:
I have implemented it but we can go back to the other method if that's preferred. I think it does make it a bit simpler in some cases but as you say it's not so complicated in general.

Reply:
Ok, if you have it already then its fine :+1: